### PR TITLE
dynamically fetch and cache asset icons based on returned asset data from Horizon

### PR DESCRIPTION
**Pull request by piyalbasu**
https://user-images.githubusercontent.com/6789586/136622445-fe786a73-f4c0-44e5-83cb-54f81a9f06ff.mov

A step by step of how icons are loaded. If the icon url is not found, the purple dot remains

### CODE REVIEW

File: src/helpers/getIconUrlFromIssuer.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import StellarSdk, { StellarTomlResolver } from "stellar-sdk";
```
Comment (piyalbasu) [inline src/helpers/getIconUrlFromIssuer.ts:1]: this is mostly lifted from Freighter, where we do something similar.

Ideally, we have an API do this for us. Maybe for v2!

File: src/pages/PoolsOverview.tsx [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+    Pools overview page
+    <div style={{ marginTop: 20 }}>
+      <Avatar
+        source={[
+          {
+            asset:
+              "USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN",
+          },
+          {
+            asset: "native",
```
Comment (piyalbasu) [inline src/pages/PoolsOverview.tsx:15]: these asset strings match what we get from Horizon:

https://horizon-testnet.stellar.org/liquidity_pools/2389378a6156eedfac66daa000d24c926431c3e667b9f754771964f27a6da6ab/

File: src/pages/PoolsOverview.tsx [hunk 3]
```diff
@@ -1,5 +1,21 @@
 import { Layout } from "@stellar/design-system";
+import { Avatar } from "components/Avatar";
```
Comment (piyalbasu) [inline src/pages/PoolsOverview.tsx:2]: will rm this after PR approval!

File: src/components/Avatar/index.tsx [hunk 4]
```diff
@@ -13,23 +16,68 @@ interface AvatarProps {
...
+    issuerKey: splitArr[1],
+  };
+};
+
+const nativeAsset = {
+  assetCode: "XLM",
+  iconUrl: StellarLogo,
+};
+
+export const Avatar = ({ source, size, borderColor }: AvatarProps) => {
```
Comment (quietbits) [inline src/components/Avatar/index.tsx:33]: Maybe we could create a new component (`AssetAvatar`, for example) that uses `Avatar` component internally? That way we could move the `Avatar` to the design system.

**Review by quietbits — APPROVED** [review]

**Pull request merged**